freeqdsk/aeqdsk.py

```diff
@@ -294,26 +596,94 @@ def write(data, fh):
             data.get("qmflag", "CLC"),
         )
     )
-    # Output data in lines of 4 values each
-    with fu.ChunkOutput(fh, chunksize=4) as output:
-        for key, default, description in fields:
-            if callable(default):
-                # Replace the default function with the value, which may depend on
-                # previously read data
-                default = default(data)
 
-            if key is None:
-                output.newline()  # Ensure on a new line
-            else:
-                output.write(data.get(key, default))
+    # Output first block of general data
+    write_array([_float_field(field, data) for field in _general_block_1], fh, data_fmt)
+
+    # Output laser bits
+    for field in _laser_block:
+        write_array(_array_field(field, data), fh, data_fmt)
+
+    # Output second block of general data
+    write_array([_float_field(field, data) for field in _general_block_2], fh, data_fmt)
+
+    # Check if we need to write an extended section
+    extended = False
+    extended_blocks = [
+        _extended_sizes,
+        _extended_arrays_1,
+        _extended_arrays_2,
+        _extended_general,
+    ]
+    for field in itertools.chain.from_iterable(extended_blocks):
+        if field.name in data:
+            extended = True
+            break
+
+    if extended:
```

Comment:
Could bail early and unindent next block?

```suggestion
    if not extended:
        return
```